allennlp/tests/data/dataset_readers/coreference/coref_reader_test.py

```diff
@@ -288,3 +288,25 @@ def check_candidate_mentions_are_well_defined(self, span_starts, span_ends, text
         # candidate spans are less than what we specified
         assert all(self.span_width >= len(x) > 0 for x in candidate_mentions)
         return candidate_mentions
+
+    def test_max_sentences(self):
+        conll_reader = ConllCorefReader(max_span_width=self.span_width)
+        instances = ensure_list(
+            conll_reader.read(str(AllenNlpTestCase.FIXTURES_ROOT / "coref" / "coref.gold_conll"))
+        )
+
+        limited_conll_reader = ConllCorefReader(max_span_width=self.span_width, max_sentences=2)
+        limited_instances = ensure_list(
+            limited_conll_reader.read(
+                str(AllenNlpTestCase.FIXTURES_ROOT / "coref" / "coref.gold_conll")
+            )
+        )
+
+        assert len(limited_instances) == len(instances) == 4
+
+        # Some are truncated, some short ones are not
+        length_of = lambda instance: len(instance.fields["text"].tokens)
```

Comment:
Can you add a check that there are no spans from other sentences as well?

Reply:
Because of the randomness it's a bit hard to test, e.g. `'X' in instance_a and 'X' not in instance_b` (if that's what you're referring to?). Alternatively I could just make the implementation to be truncation instead of sampling -- I don't think that makes too much of a difference anyway.

Reply:
EDIT: changed to truncation.